jykuo-love-shiritori/twp
# [frontend] Cart api

Reeeeeeeeeeeeesolves #94 

### Changes
- [x] [/buyer/cart](http://localhost:8080/buyer/cart)
- cart page apis

#### Function
- carts section
  - change quantity of product from 0 to that stock of product
  - beware of that if quantity is equal 0 it will remove that row entirely and remove whole cart if there's no product in it
  - if there's any product in the cart is not availabe, it will alert and tell u what's went wrong
- checkout section (the canva)
  - see the detailed data
  - add/remove used coupons
  - select credit card
  - pay. will check if any card is selected and when succeed, link back to carts
- useable coupon section (the modal)
  - see the usable coupons
  - click one to use it 

### Testing Approach (if unconventional)
```sh
docker compose up --build
```
- login as customer(user)
- click and add product to cart in home page or from shop
- go to [/buyer/cart](http://localhost:8080/buyer/cart)
- try to checkout, add/remove usable coupon
- try to pay the cart
- go to [order history](http://localhost:8080/user/buyer/order) to see if the cart u paid is there
